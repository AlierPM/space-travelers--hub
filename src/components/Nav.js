import React from 'react';
import { NavLink } from 'react-router-dom';
import '../App.css';
import pic from '../img/planet.png';
import RouterNavLink from './Router';

function Nav() {
  return (
    <div className="main">
      <div className="navbarMenu">
        <div className="logo">
          <img className="img" src={pic} alt="saturn" />
          <h1>Space Travelers Hub</h1>
        </div>
        <div className="menu">
          <ul>
            <li>
              <NavLink to="/rockets">Rockets</NavLink>
            </li>
            <li>
              <NavLink to="/missions">Missions</NavLink>
            </li>
            <li>
              <NavLink to="/dragons">Dragons</NavLink>
            </li>
            <li>
              <NavLink end to="/">
                My Profile
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
      <RouterNavLink />
    </div>
  );
}
export default Nav;
