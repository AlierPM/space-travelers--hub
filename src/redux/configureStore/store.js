import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import dragonReducer from "../dragon/dragon";
import rocketSlice from "../rocket/rocketSlice";
import missionsReducer from "../mission/missionSlice";

const store = configureStore({
  reducer: {
    dragon: dragonReducer,
    rockets: rocketSlice,
    missions: missionsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
