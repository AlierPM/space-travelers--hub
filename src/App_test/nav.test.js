import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Nav from '../components/Nav';

test('testing prolife', () => {
  const nav = render(
    <React.StrictMode>
      <Provider store={store}>
        <Router>
          <Nav />
        </Router>
      </Provider>
    </React.StrictMode>,
  );
  expect(nav).toMatchSnapshot();
});
