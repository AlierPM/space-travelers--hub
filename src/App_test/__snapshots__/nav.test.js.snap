// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing prolife 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="navbarMenu"
        >
          <div
            class="logo"
          >
            <img
              alt="saturn"
              class="img"
              src="planet.png"
            />
            <h1>
              Space Travelers Hub
            </h1>
          </div>
          <div
            class="menu"
          >
            <ul>
              <li>
                <a
                  class=""
                  href="/rockets"
                >
                  Rockets
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="displayMission"
          >
            <h1>
               My Missions
            </h1>
            <p>
              No Missions joined 
            </p>
          </div>
          <div
            class="availableRockets"
          >
            <h1>
              My Rockets
            </h1>
            <p>
              No rockets reserved
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="navbarMenu"
      >
        <div
          class="logo"
        >
          <img
            alt="saturn"
            class="img"
            src="planet.png"
          />
          <h1>
            Space Travelers Hub
          </h1>
        </div>
        <div
          class="menu"
        >
          <ul>
            <li>
              <a
                class=""
                href="/rockets"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="displayMission"
        >
          <h1>
             My Missions
          </h1>
          <p>
            No Missions joined 
          </p>
        </div>
        <div
          class="availableRockets"
        >
          <h1>
            My Rockets
          </h1>
          <p>
            No rockets reserved
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
